lib/LaTeXML/Package/physics.sty.ltxml

```diff
@@ -0,0 +1,630 @@
+#  Optional exponent: a standard LaTeX optional argument, [], can be given
+#     for an extra optional power(-like) argument.
+#     BUT: for some macros (trigs) that ONLY works if a delimited argument is given.
+
+#----------------------------------------------------------------------
+# Parsing aids
+our %physics_delimiters = (
+  '(' => T_OTHER(')'),
+  '[' => T_OTHER(']'),
+  '|' => T_OTHER('|'),
+);    # And probably more...
+
+# returns one of (0,1,Token) meaning (no stretch, stretchy, specific size)
+sub phys_readDelimiterSize {
+  my ($gullet) = @_;
+  my $size     = 1;
+  my $next     = $gullet->readToken();    # or XToken ?
+  if ($next->equals(T_OTHER('*'))) {      # No stretch
+    $size = 0;
+    $next = $gullet->readToken(); }       # or XToken ?
+  if (ToString($next) =~ /^\\(b|B)igg?/) {    # \big,\Big,\bigg,\Bigg
+    $size = $next;
+    $next = $gullet->readToken(); }           # or XToken ?
+  $gullet->unread($next);
+  return $size; }
+
+sub phys_open {
+  my ($size, $delim) = @_;
+  return ($delim && $delim->unlist
+    ? (ref $size ? Tokens($size, $delim) : ($size ? Tokens(T_CS('\left'), $delim) : $delim))
+    : ()); }
+
+sub phys_mid {
+  my ($size, $delim) = @_;
+  return ($delim && $delim->unlist
+    ? (ref $size ? Tokens($size, $delim) : ($size ? Tokens(T_CS('\middle'), $delim) : $delim))
+    : ()); }
+
+sub phys_close {
+  my ($size, $delim) = @_;
+  return ($delim && $delim->unlist
+    ? (ref $size ? Tokens($size, $delim) : ($size ? Tokens(T_CS('\right'), $delim) : $delim))
+    : ()); }
+
+# Read a single arg, either TeX delimited {}, or delmited according to hash
+# Error is signaled if $required and no arg found
+# $open,$close are the fences found, or passed through for TeX {}
+sub phys_readArg {
+  my ($gullet, $required, $open, $close, %delimiters) = @_;
+  my $arg;
+  my $next = $gullet->readToken();    # or XToken ?
+  if ($next && $next->equals(T_BEGIN)) {    # Read regular arg, use default delimiters
+    $gullet->unread($next);
+    $arg = $gullet->readArg(); }
+  elsif (my $c = $next && $delimiters{ ToString($next) }) {
+    $open = $next; $close = $c;
+    $arg  = $gullet->readUntil($c); }
+  else {
+    $gullet->unread($next);
+    Error('unexpected', $next, $gullet, "Expected an open delimiter", "Got " . Stringify($next))
+      if $required; }
+  return (wantarray ? ($arg, $open, $close) : $arg); }
+
+#----------------------------------------------------------------------
+# Construction aids
+
+# Note: This could be applicative-power ???
+sub phys_applyPower {
+  my ($cfunc, $pfunc, $power) = @_;
+  my ($cargs, $pargs) = dualize_arglist('#1', $power);
+  if ($cfunc) {
+    $cfunc = Invocation(T_CS('\@APPLY'),
+      Tokens(Invocation(T_CS('\@CSYMBOL'), T_OTHER('power')),
+        $cfunc, $$cargs[0])); }
+  if ($pfunc) {
+    $pfunc = Tokens($pfunc, Invocation(T_CS('\@@POSTSUPERSCRIPT'), $$pargs[0])); }
+  return ($cfunc, $pfunc); }
```

Comment:
I see how this ties nicely into the XMDual experimentation I am doing at the a11y-experiment branch. Won't repeat any feedback from there, interesting read to see how you're using the dual-related macros in action!